templates/todo/task_list.html

```diff
@@ -0,0 +1,68 @@
+{% extends "base.html" %}
+
+{% block content %}
+    <h1>TODO list</h1>
+    <button style="float: right; bottom: 50px; position: relative" class="btn btn-primary"
+            type="submit"
+            value="submit">
+        <a role="button" style="color: white"
+           href="{% url 'todo:task-create' %}">Add task</a>
+    </button>
```

Comment:
Using `a` tag with class `btn` will render a button with redirect on a whole element, in that case you need to click on text to get to the given page.

Also try avoiding inline css, create classes for elements and write css in separate files.